application/frontend/tests/craft.test.ts

```diff
@@ -0,0 +1,159 @@
+import { test, Page } from '@playwright/test';
+
+async function getGameCanvas(page: Page) {
+  const canvas = await page.waitForSelector('#game-container canvas', { timeout: 10000 });
+  await page.waitForTimeout(2000);
+  return canvas;
+}
+
+function calculateScreenPosition(
+  position: { x: number; y: number },
+  bounds: { x: number; y: number },
+  scale: { x: number; y: number },
+  zoomLevel: number
+) {
+  return {
+    x: bounds.x + (position.x * scale.x * zoomLevel),
+    y: bounds.y + (position.y * scale.y * zoomLevel)
+  };
+}
+
+
+async function attemptClicksAtPosition(
+  page: Page,
+  screenX: number,
+  screenY: number,
+  mapX: number,
+  mapY: number,
+  attempts: number = 5,
+  delayBetweenClicks: number = 300
+) {
+  for (let attempt = 1; attempt <= attempts; attempt++) {
+    console.log(`Click attempt ${attempt} at map position (${mapX}, ${mapY}) -> screen position (${Math.round(screenX)}, ${Math.round(screenY)})`);
+    test.setTimeout(100000);
+    await page.mouse.click(screenX, screenY);
+    await page.waitForTimeout(delayBetweenClicks);
+  }
+}
+
+test.describe('Game Resource Collection and Crafting', () => {
+  const humanPositions = [
+    // human_x positions
+    { x: 176, y: 192 },
+    { x: 256, y: 352 },
+    { x: 192, y: 368 },
+    { x: 272, y: 48 },
+    { x: 336, y: 112 },
+    // human_y positions
+    { x: 448, y: 96 },
+    { x: 112, y: 112 },
+    { x: 448, y: 416 }
+  ];
+
+  test('complete game workflow', async ({ page }) => {
+    test.setTimeout(100000);
+    await page.goto('/game');
+    const canvas = await getGameCanvas(page);
+    const bounds = await canvas.boundingBox();
+
+    if (!bounds) {
+      throw new Error('Could not get canvas bounds');
+    }
+
+    const displaySize = await page.evaluate(() => {
+      const canvas = document.querySelector('#game-container canvas');
+      if (!canvas) return null;
+      return {
+        width: canvas.clientWidth,
+        height: canvas.clientHeight
+      };
+    });
+
+    console.log('Canvas display size:', displaySize);
+    if (!displaySize) {
+      throw new Error('Could not get canvas display size');
+    }
+
+    const scale = {
+      x: displaySize.width / 2048,
+      y: displaySize.height / 1152
+    };
+
+    // 1. Click humans
```

Comment:
For each of these steps, could you break them up into individual `test()` calls? This will annotate each of the step as the tests are running and can provide better context when they fail.